From 5d04ab83fccacb600bba2bccb8f42ecce59bd044 Mon Sep 17 00:00:00 2001
From: Bert van Hall <bert.vanhall@avionic-design.de>
Date: Tue, 3 Nov 2015 13:24:43 +0100
Subject: [PATCH 2/2] cross-compile: do not add LIBDIR and INCLUDEDIR to flags

Do not add LIBDIR and INCLUDEDIR to c resp. ld flags as those are
targets for installation only and point to the build host's system. This
leads to linker errors and wrong headers being included.

Signed-off-by: Bert van Hall <bert.vanhall@avionic-design.de>
---
 setup.py | 11 -----------
 1 file changed, 11 deletions(-)

diff --git a/setup.py b/setup.py
index da67731..2375189 100644
--- a/setup.py
+++ b/setup.py
@@ -510,17 +510,6 @@ class PyBuildExt(build_ext):
                     for directory in reversed(options.dirs):
                         add_dir_to_list(dir_list, directory)
 
-        if os.path.normpath(sys.base_prefix) != '/usr' \
-                and not sysconfig.get_config_var('PYTHONFRAMEWORK'):
-            # OSX note: Don't add LIBDIR and INCLUDEDIR to building a framework
-            # (PYTHONFRAMEWORK is set) to avoid # linking problems when
-            # building a framework with different architectures than
-            # the one that is currently installed (issue #7473)
-            add_dir_to_list(self.compiler.library_dirs,
-                            sysconfig.get_config_var("LIBDIR"))
-            add_dir_to_list(self.compiler.include_dirs,
-                            sysconfig.get_config_var("INCLUDEDIR"))
-
         # lib_dirs and inc_dirs are used to search for files;
         # if a file is found in one of those directories, it can
         # be assumed that no additional -I,-L directives are needed.
-- 
2.1.0

